### PR TITLE
fix(framework): Add missing `peerDependencies` and fix dynamic imports

### PR DESCRIPTION
**Pull request by rifont**
### What changed? Why was the change needed?
* Specify missing `peerDepencies`. Without specifying `peerDepencies`, Rollup (via TS-UP) bundles all deps in `devDependencies` into the bundle, which significantly increases bundle size. New bundle sizes below:
* Import dependencies in statically analyzable way (fix the `cross-fetch` and `zod-to-json-schema` imports)

### Screenshots

_New bundle size after specifying peer deps (before, all CJS files were of similar size due to `zod` being bundled)_
```bash
DTS Build start
CJS dist/h3.js        23.34 KB
CJS dist/nuxt.js      23.35 KB
CJS dist/remix.js     23.29 KB
CJS dist/index.js     111.20 KB
CJS dist/next.js      24.14 KB
CJS dist/sveltekit.js 23.55 KB
CJS dist/express.js   23.41 KB
CJS ⚡️ Build success in 228ms
DTS ⚡️ Build success in 3380ms
DTS dist/index.d.ts            10.33 KB
DTS dist/h3.d.ts               427.00 B
DTS dist/express.d.ts          308.00 B
DTS dist/next.d.ts             1.19 KB
DTS dist/nuxt.d.ts             427.00 B
DTS dist/sveltekit.d.ts        561.00 B
DTS dist/remix.d.ts            383.00 B
DTS dist/handler-PIXNhv7t.d.ts 194.38 KB
```

_Next.js static analysis import error (FIXED)_
```bash
Critical dependency: require function is used in a way in which dependencies cannot be statically extracted

Import trace for requested module:
../novu/packages/framework/dist/next.js
./app/api/novu/route.ts
```

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: packages/framework/package.json [hunk 1]
```diff
@@ -86,13 +86,48 @@
...
+    "zod": {
+      "optional": true
+    },
+    "zod-to-json-schema": {
+      "optional": true
+    }
+  },
   "devDependencies": {
-    "@sveltejs/kit": "^2.5.10",
+    "@sveltejs/kit": "^1.27.3",
```
Comment (rifont) [inline packages/framework/package.json:125]: Pinning back to the minimum supported version

File: packages/framework/src/next.ts [hunk 2]
```diff
@@ -24,7 +24,13 @@ export const frameworkName: SupportedFrameworkName = 'next';
...
  * ```
  */
-export const serve = (options: ServeHandlerOptions): any => {
+export const serve = (
+  options: ServeHandlerOptions
+): ((expectedReq: NextRequest, res: NextApiResponse) => Promise<Response>) & {
+  GET: (expectedReq: NextRequest, res: NextApiResponse) => Promise<Response>;
+  POST: (expectedReq: NextRequest, res: NextApiResponse) => Promise<Response>;
+  OPTIONS: (expectedReq: NextRequest, res: NextApiResponse) => Promise<Response>;
+} => {
```
Comment (rifont) [inline packages/framework/src/next.ts:33]: Adding strong typings to the Next.js serve function

File: packages/framework/src/next.ts [hunk 3]
```diff
@@ -179,12 +185,10 @@ export const serve = (options: ServeHandlerOptions): any => {
   const handlerFunctions = Object.defineProperties(defaultHandler, {
     GET: { value: baseHandler.bind(null, 'GET') },
     POST: { value: baseHandler.bind(null, 'POST') },
-    PUT: { value: baseHandler.bind(null, 'PUT') },
```
Comment (rifont) [inline packages/framework/src/next.ts:182]: This isn't used today, removing for now.

File: packages/framework/src/remix.ts [hunk 4]
```diff
@@ -18,15 +19,7 @@ export const serve = (
...
           // eslint-disable-next-line @typescript-eslint/naming-convention
-          let Res: typeof Response;
-
-          if (typeof Response === 'undefined') {
-            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-var-requires
-            Res = require('cross-fetch').Response;
-          } else {
-            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-            Res = Response;
-          }
```
Comment (rifont) [inline packages/framework/src/remix.ts:29]: This logic has a util to reuse.

**Comment by scopsy** [comment]
Amazing!

**Review by scopsy — APPROVED** [review]

**Pull request merged**